### PR TITLE
Checks for left_index and right_index merge parameters

### PR DESCRIPTION
**Pull request by ivallesp**
Hi,

I just committed an error when I was doing an analysis using pandas and this motivated me to implement two checks which in my opinion are necessary. 

I was trying to perform a merge and I confused the parameters "left_on" and "right_on" for "left_index" and "right_index". I ran the code and it did not raised me any error. It produced a table which seem to be fine in terms of shape. I suspect that what happened is that the tables got merged by the index of the data frame. I think it would be a great idea to check if right_index and left_index are of type bool, if not, raise an error. This way we will avoid that more people got the same error as mine :D.

Tests passed 

Thanks!

### CODE REVIEW

File: pandas/tools/merge.py [hunk 1]
```diff
@@ -471,6 +471,14 @@ def __init__(self, left, right, how='inner', on=None,
             raise ValueError(
                 'can not merge DataFrame with instance of '
                 'type {0}'.format(type(right)))
+        if not isinstance(left_index, bool):
```
Comment (sinhrks) [inline pandas/tools/merge.py]: pls use ``pandas.api.types.is_bool``.

File: pandas/tools/merge.py [hunk 2]
```diff
@@ -471,6 +471,14 @@ def __init__(self, left, right, how='inner', on=None,
...
+            raise ValueError(
+                'left_index parameter must be of type bool, not '
```
Comment (sinhrks) [inline pandas/tools/merge.py]: can u add tests to check expected error is raised?

File: pandas/tools/merge.py (hunk 1 above)
Comment (ivallesp) [inline pandas/tools/merge.py]: done

File: pandas/tools/merge.py (hunk 2 above)
Comment (ivallesp) [inline pandas/tools/merge.py]: done, thanks!

File: pandas/tests/test_categorical.py [hunk 3]
```diff
@@ -4049,6 +4049,24 @@ def test_merge(self):
         result = pd.merge(cleft, cright, how='left', left_on='b', right_on='c')
         tm.assert_frame_equal(result, expected)
 
+        # params left_index right_index checks
+        self.assertRaises(ValueError,
```
Comment (jreback) [inline pandas/tests/test_categorical.py]: wrong file. should go in tests/tools/test_merge.py

**Comment by ivallesp** [comment]
Updated!

**Comment by ivallesp** [comment]
Is there still any change to do? Thanks!

File: pandas/tools/tests/test_merge.py [hunk 4]
```diff
@@ -109,6 +109,15 @@ def test_merge_misspecified(self):
         self.assertRaises(ValueError, merge, self.df, self.df2,
                           left_on=['key1'], right_on=['key1', 'key2'])
 
+    def test_index_and_on_parameters_confusion(self):
+        self.assertRaises(ValueError, merge, self.df, self.df2, how='left',
```
Comment (jreback) [inline pandas/tools/tests/test_merge.py]: add the github issue reference as a comment

**Comment by jreback** [comment]
pls add a whatsnew entry in 0.19.1. ping on green.

**Comment by ivallesp** [comment]
Done!

**Comment by ivallesp** [comment]
@jreback green

File: doc/source/whatsnew/v0.19.1.txt [hunk 5]
```diff
@@ -25,6 +25,13 @@ Performance Improvements
 
 
 
+.. _whatsnew_0191.new_features:
+
+New features
+~~~~~~~~~~~~
+- Add checks to assure that left_index and right_index are of type bool
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.19.1.txt]: Can you move this to the bug fixes section? (I think that it is rather a bug that it accepted a wrong value, than a new feature that it now checks that)

File: pandas/tools/tests/test_merge.py (hunk 4 above)
Comment (jorisvandenbossche) [inline pandas/tools/tests/test_merge.py]: Can you address this one?

**Review by jorisvandenbossche — COMMENTED** [review]
@ivallesp two small comments, looks good!

**Comment by jorisvandenbossche** [comment]
@ivallesp No matter, @jreback already merged and addressed those comments!

**Comment by jreback** [comment]
@jorisvandenbossche was your 2nd one the comment reference in the tests? (decided not a big deal)

**Comment by jorisvandenbossche** [comment]
yep, but indeed not a big deal

**Pull request closed without merging**